File: frontend/src/api/job.ts
/**
 * 职位相关API
 */
import { http, ApiResponse, PageResponse } from '@/utils/request'

// 职位接口
export interface Job {
  id: string
  job_name: string
  company_name: string
  salary: string
  city: string
  area?: string
  experience?: string
  education?: string
  company_size?: string
  industry?: string
  job_url: string
  boss_title?: string
  status: string
  is_applied: boolean
  notes?: string
  created_at: string
  updated_at: string
}

// 职位查询参数
export interface JobQuery {
  page?: number
  page_size?: number
  city?: string
  keyword?: string
}

// 创建职位参数
export interface JobCreate {
  job_name: string
  company_name: string
  salary: string
  city: string
  area?: string
  experience?: string
  education?: string
  company_size?: string
  industry?: string
  job_url: string
  boss_title?: string
}

// 更新职位参数
export interface JobUpdate {
  status?: string
  is_applied?: boolean
  notes?: string
}

export const jobApi = {
  // 获取职位列表
  getList(params: JobQuery) {
    return http.get<PageResponse<Job>>('/jobs', { params })
  },

  // 获取职位详情
  getDetail(id: string) {
    return http.get<Job>(`/jobs/${id}`)
  },

  // 创建职位
  create(data: JobCreate) {
    return http.post<Job>('/jobs', data)
  },

  // 更新职位
  update(id: string, data: JobUpdate) {
    return http.put<Job>(`/jobs/${id}`, data)
  },

  // 删除职位
  delete(id: string) {
    return http.delete(`/jobs/${id}`)
  },

  // 批量创建职位
  batchCreate(data: JobCreate[]) {
    return http.post<{ count: number }>('/jobs/batch', data)
  },
}

File: backend/app/schemas/job.py
"""
职位相关数据模型
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class JobBase(BaseModel):
    """职位基础模型"""

    job_name: str = Field(..., description="职位名称")
    company_name: str = Field(..., description="公司名称")
    salary: str = Field(..., description="薪资范围")
    city: str = Field(..., description="城市")
    area: Optional[str] = Field(None, description="区域")
    experience: Optional[str] = Field(None, description="经验要求")
    education: Optional[str] = Field(None, description="学历要求")
    company_size: Optional[str] = Field(None, description="公司规模")
    industry: Optional[str] = Field(None, description="行业")
    job_url: str = Field(..., description="职位链接")
    boss_title: Optional[str] = Field(None, description="BOSS职位")


class JobCreate(JobBase):
    """创建职位"""

    pass


class JobUpdate(BaseModel):
    """更新职位"""

    status: Optional[str] = Field(None, description="状态")
    is_applied: Optional[bool] = Field(None, description="是否已投递")
    notes: Optional[str] = Field(None, description="备注")


class JobResponse(JobBase):
    """职位响应"""

    id: int = Field(..., description="职位ID")
    status: str = Field(default="pending", description="状态")
    is_applied: bool = Field(default=False, description="是否已投递")
    notes: Optional[str] = Field(None, description="备注")
    created_at: datetime = Field(default_factory=datetime.now, description="创建时间")
    updated_at: datetime = Field(default_factory=datetime.now, description="更新时间")

    class Config:
        json_schema_extra = {
            "example": {
                "id": 1,
                "job_name": "Python开发工程师",
                "company_name": "科技公司",
                "salary": "15-25K",
                "city": "北京",
                "status": "pending",
                "is_applied": False,
            }
        }


class JobFilter(BaseModel):
    """职位筛选"""

    city: Optional[str] = None
    keyword: Optional[str] = None
    experience: Optional[str] = None
    education: Optional[str] = None
    salary_min: Optional[int] = None
    salary_max: Optional[int] = None
